pkg/virt-api/rest/subresource.go

```diff
@@ -1545,3 +1555,261 @@ func (app *SubresourceAPIApp) RemoveMemoryDumpVMRequestHandler(request *restful.
 
 	response.WriteHeader(http.StatusAccepted)
 }
+
+func generateVMIInterfaceRequestPatch(vmi *v1.VirtualMachineInstance, interfaceRequest *v1.VirtualMachineInterfaceRequest) (string, error) {
+	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+		if alreadyExists(interfaceRequest, iface) {
+			return "", fmt.Errorf("unable to add interface [%s] because it already exists", iface.Name)
```

Comment:
Why would we care if someone tries to hotplug an existing interface?
If it is already there, then we can just ignore it. No?
Or is there a reason for not following idempotence in this case?